client/rest-high-level/src/main/java/org/elasticsearch/client/Request.java

```diff
@@ -597,6 +598,18 @@ static Request rollover(RolloverRequest rolloverRequest) throws IOException {
         return new Request(HttpPost.METHOD_NAME, endpoint, params.getParams(), entity);
     }
 
+    static Request getSettings(GetSettingsRequest getSettingsRequest) throws IOException {
+        Params params = Params.builder();
+        params.withIndicesOptions(getSettingsRequest.indicesOptions());
+        params.withLocal(getSettingsRequest.local());
+
+        if (getSettingsRequest.indices().length == 0) {
+            throw new IllegalArgumentException("getSettings requires at least one index");
+        }
```

Comment:
I think this validation is not needed: we also register the `/_settings/{name}` that doesn't require any index. We need to make the client go to that one endpoint if no indices are provided.